lib/baw-workers/redis_communicator.rb

```diff
@@ -0,0 +1,122 @@
+module BawWorkers
+  class DecodeException < StandardError;
+  end
+
+  class RedisCommunicator
+
+    attr_accessor :logger
+
+    # Create a new BawWorkers::ApiCommunicator.
+    # @param [Object] logger
+    # @param [Object] redis_instance
+    # @param [Object] settings
+    # @return [BawWorkers::RedisCommunicator]
+    def initialize(logger, redis_instance, settings = {})
+      @logger = logger
+      @redis = redis_instance
+      @settings = settings
+
+      @class_name = self.class.name
+    end
+
+    def namespace
+      @settings[:namespace] || 'baw-workers'
+    end
+
+    def add_namespace(key)
+      namespace + ':' + key
+    end
+
+    def redis
+      @redis
+    end
+
+
+    # @param [String] key
+    # @return [Boolean]
+    def delete(key, opts = {})
+      key = add_namespace(key) unless opts[:no_namespace]
+
+      deleted = @redis.del(key)
+
+      raise 'Too many keys deleted' if deleted > 1
```

Comment:
I'm glad this is checked, but isn't after it deletes too many keys a bit late? Should this do a get first to check the key count?